test/sql/copy/csv/test_union_by_name.test

```diff
@@ -0,0 +1,112 @@
+# name: test/sql/copy/csv/test_union_by_name.test
+# description: UNION_BY_NAME test
+# group: [csv]
+
+statement ok
```

Comment:
Could we also add a test that verifies that types are combined in a sane manner, e.g. `int32 + int64 = int64`, etc?

We could do this using the `typeof` function:

```sql
SELECT typeof(i::INT64) FROM range(10) tbl(i) LIMIT 1;
-- BIGINT
```

Reply:
I add some new test for types check